File: library/src/main/java/tw/shounenwind/plurkconnection/callbacks/ApiNullCallback.java
package tw.shounenwind.plurkconnection.callbacks;

import okhttp3.Response;
import okhttp3.ResponseBody;
import tw.shounenwind.plurkconnection.responses.ApiResponseNull;

public abstract class ApiNullCallback extends BasePlurkCallback<ApiResponseNull> {

    @Override
    public void runResult(Response response) throws Exception {
        super.runResult(response);
        ResponseBody body = response.body();
        if (body != null)
            body.close();
        onSuccess(new ApiResponseNull(response.code(), null));
    }
}

File: library/src/main/java/tw/shounenwind/plurkconnection/callbacks/ApiNullCallback.kt
package tw.shounenwind.plurkconnection.callbacks

import okhttp3.Response
import tw.shounenwind.plurkconnection.responses.ApiResponseNull

class ApiNullCallback : BasePlurkCallback<ApiResponseNull>() {

    @Throws(Exception::class)
    override fun onSuccess(parsedResponse: ApiResponseNull) {

    }

    @Throws(Exception::class)
    override fun runResult(response: Response) {
        super.runResult(response)
        val body = response.body()
        body?.close()
        onSuccess(ApiResponseNull(response.code()))
    }
}
